src/sid/simulate.py

```diff
@@ -961,4 +981,71 @@ def _add_additional_information_to_states(
     if n_has_additionally_infected is not None:
         states["n_has_infected"] += n_has_additionally_infected
 
+    if "infection_probability_multiplier" in columns_to_keep:
+        states["infection_probability_multiplier"] = infection_probability_multiplier
+
     return states
+
+
+def _prepare_infection_probability_multiplier(
+    infection_probability_multiplier_model: Optional[Callable],
+    initial_states: pd.DataFrame,
+    params: pd.DataFrame,
+    seed: int,
+) -> np.ndarray:
+    """Prepare the multiplier for infection probabilities.
+
+    The multiplier defines individual susceptibility which can be used to let infection
+    probabilities vary by age.
+
+    If not multiplier is given, all individuals have the same susceptibility. Otherwise,
+    a custom function generates multipliers for the infection probability for each
+    individual.
+
+    Args:
+        infection_probability_multiplier_model (Optional[Callable]): The custom function
+            which computes individual multipliers with states, parameters and a seed.
+        initial_states (pandas.DataFrame): The initial states.
+        params (pandas.DataFrame): The parameters.
+        seed (int): An integer which can be used by the user for reproducibility.
+
+    Returns: infection_probability_multiplier (numpy.ndarray): An array with a
+        multiplier for each individual between 0 and 1.
+
+    """
+    if infection_probability_multiplier_model is None:
+        infection_probability_multiplier = np.ones(len(initial_states))
```

Comment:
It would be more consistent with the rest of sid if we keep the `infection_probability_multiplier` in a `pandas.Series` for most of the time. Only when calling the numba function we would convert it. But I don't have a strong opinion here.

Reply:
We have a lot of objects which are np.ndarrays or the same objects in numba lists and pandas objects. I do not think it would be less consistent. And, I would have to call to_numpy later anyway. Then, I prefer to create the correct object here and never touch it later.